### DISCUSSION TITLE
Itowns , BIM model and 3D Object

### DISCUSSION THREAD

**Discussion by sis-irenee** [iTowns/itowns#1663] [Q&A]
Hello  everybody,
I am currently working, as part of my end-of-study project, on the integration of a BIM model and a point cloud in a web GIS environment. We work with Revit (for the BIM part) and Itowns (for the web part).

I converted my mockup to FBX (from Révit) and then converted it to DAE (with an online converter) to be able to view it in Itowns.
However, I encountered some problems when integrating my model :

- the model is much too large, so I have to apply a scale factor to have it the right size,
- Georeferencing level, I have to do it manually so that the model is in the right position. However, there is a risk of loss of precision,
- Finally, I want to be able to query elements of my model but that is impossible with this format because it does not contain any information such as for example a 3dtiles.

I have found a few alternatives but they don't really work well.

my question is:

- how to automatically georeference a 3D object in Itowns (or at worst damage it)
- Is there another more suitable format for viewing BIM in Itowns ?

NB: I am currently processing the passage IFC-CityGML- 3D tiles.

I've been struggling a bit with it for 2 months, so any help will be welcome (link, techniques, ideas, etc).
thank you so much.

![itowns discuss](https://user-images.githubusercontent.com/82024431/119788333-572e6500-bed2-11eb-9d6d-daee1985a01b.JPG)
[exemple.txt](https://github.com/iTowns/itowns/files/6552165/exemple.txt)

**Comment by gchoqueux** [comment]
Hello,

You could use [`FBXLoader`](https://github.com/mrdoob/three.js/blob/dev/examples/js/loaders/FBXLoader.js) instead of Collada loader.

> how to automatically georeference a 3D object in Itowns (or at worst damage it)

For `GlobeView`, you have to convert the projection to `EPSG:4978`.
You could use `PlanarView` if your original projection is conic (Lambert)

> Is there another more suitable format for viewing BIM in Itowns 

3D tiles is the best format


I come back to you to give you a process to convert to 3d tiles

**Reply by sis-irenee** [reply]
Hello,
thank you for the answer, I will test on my model.

I am going to test FBXLoader, because with Collada I have to apply a scaling factor.

my model is georeferenced in RGF93 CC49.

ok I await your feedback regarding 3D tiles
